ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/metadata/MetadataPresenterImpl.java

```diff
@@ -217,6 +222,16 @@ public static Avu newMetadata(String attr, String value, String unit) {
         return avu;
     }
 
+    @Override
+    public Avu  setAvuModelKey(Avu avu) {
+        if (avu != null) {
+            final AutoBean<Avu> avuBean = AutoBeanUtils.getAutoBean(avu);
+            avuBean.setTag(AVU_BEAN_TAG_MODEL_KEY, String.valueOf(unique_avu_id++));
+            return avuBean.as();
```

Comment:
The biggest reason I see for not doing that is the method "getKey" of ModelKeyProvider will causing "side-effect" that is not expected. The method is intended for just getting the keys. 
Also If the key does not exist, I would like the code throw a exception because, as far the logic goes, the every item in the store should have a tag. So if indeed there is an assertion error because of no tag, it will show up here.